src/createServer.js

```diff
@@ -1,3 +1,67 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+const PORT = 5700;
+
+const caseNames = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+const ERROR_MESSAGES = {
+  NO_ORIGIN_STRING: 'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+  NO_QUERY_CASE: '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+  NO_SUPPORTED_CASE: 'This case is not supported. Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+};
+
+const checkValidation = (textToTransform, toCase) => {
+  const errors = [];
+
+  if (!textToTransform) {
+    errors.push({ message: ERROR_MESSAGES.NO_ORIGIN_STRING });
+  }
+
+  if (!toCase) {
+    errors.push({ message: ERROR_MESSAGES.NO_QUERY_CASE });
+  }
+
+  if (toCase && !caseNames.includes(toCase)) {
+    errors.push({ message: ERROR_MESSAGES.NO_SUPPORTED_CASE });
+  }
+
+  return errors;
+};
+
+const createServer = () => {
+  const server = http.createServer((request, response) => {
+    response.setHeader('Content-type', 'application/json');
+
+    const [originString, queryCase] = request.url.slice(1).split('?');
+
+    const toCase = queryCase?.split('=')[1];
```

Comment:
Nitpick:
I am pretty sure your solution works as well. However, good practice is to use `URLSeatchParams` :
```
 const searchParams = new URLSearchParams(queryToCase);
 const targetCase = searchParams.get('toCase');
```